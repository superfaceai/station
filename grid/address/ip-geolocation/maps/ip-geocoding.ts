/* eslint-disable jest/no-export */
import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

import { buildSuperfaceTest } from '../../../test-config';

export const ipGeolocationTest = (
  provider: string,
  recordingOptions?: RecordingProcessOptions
): void => {
  describe(`address/ip-geolocation/${provider}`, () => {
    let superfaceTest: SuperfaceTest;

    beforeEach(() => {
      superfaceTest = buildSuperfaceTest({
        profile: 'address/ip-geolocation',
        provider,
      });
    });

    describe('IpGeolocation', () => {
      describe('when api key is valid', () => {
        it('should return geolocation coordinates and address for valid IP address', async () => {
          await expect(
            superfaceTest.run(
              {
                useCase: 'IpGeolocation',
                input: { ipAddress: '8.8.8.8' },
              },
              recordingOptions
            )
          ).resolves.toMatchSnapshot();
        });

        it('should return client IP geolocation coordinates when no IP address specified in input', async () => {
          await expect(
            superfaceTest.run(
              {
                useCase: 'IpGeolocation',
                input: {},
              },
              recordingOptions
            )
          ).resolves.toMatchSnapshot();
        });

        it('should return bad request error when IP address format is wrong', async () => {
          await expect(
            superfaceTest.run(
              {
                useCase: 'IpGeolocation',
                input: { ipAddress: '256.256.256.256' },
              },
              recordingOptions
            )
          ).resolves.toMatchSnapshot();
        });
      });
    });
  });
};
