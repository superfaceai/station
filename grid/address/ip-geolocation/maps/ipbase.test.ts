import { ipGeolocationTest } from './ip-geocoding';

ipGeolocationTest('ipbase');
