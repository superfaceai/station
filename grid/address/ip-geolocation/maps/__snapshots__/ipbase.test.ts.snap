// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`address/ip-geolocation/ipbase IpGeolocation when api key is valid should return bad request error when IP address format is wrong 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Bad request\\",
  \\"detail\\": \\"The ip must be a valid IP address.\\"
}
AST Path: definitions[0].statements[2].responseHandlers[1].statements[1]
Original Map Location: Line 51, column 7",
}
`;

exports[`address/ip-geolocation/ipbase IpGeolocation when api key is valid should return client IP geolocation coordinates when no IP address specified in input 1`] = `
Ok {
  "value": Object {
    "addressCountry": "Czechia",
    "addressCountryCode": "CZ",
    "addressLocality": "Jablonec nad Nisou",
    "addressRegion": "Libereck√Ω kraj",
    "ip": "94.127.131.198",
    "latitude": 50.68013000488281,
    "longitude": 15.321399688720703,
    "postalCode": "468 22",
    "timeZone": "Europe/Prague",
  },
}
`;

exports[`address/ip-geolocation/ipbase IpGeolocation when api key is valid should return geolocation coordinates and address for valid IP address 1`] = `
Ok {
  "value": Object {
    "addressCountry": "United States",
    "addressCountryCode": "US",
    "addressLocality": "Mountain View",
    "addressRegion": "California",
    "ip": "8.8.8.8",
    "latitude": 37.38801956176758,
    "longitude": -122.07431030273438,
    "postalCode": "94041",
    "timeZone": "America/Los_Angeles",
  },
}
`;
